vernac/classes.ml

```diff
@@ -546,8 +571,36 @@ let new_instance ?(global=false) ~program_mode
       Namegen.next_global_ident_away i (Termops.vars_of_env env)
   in
   let env' = push_rel_context ctx env in
+  id, env', sigma, k, u, cty, ctx', ctx, imps, subst, decl
+
+let new_instance_interactive ?(global=false) ~program_mode
+    poly instid ctx cl
+    ?(generalize=true) ?(tac:unit Proofview.tactic option) ?hook pri =
+  let env = Global.env() in
+  let id, env', sigma, k, u, cty, ctx', ctx, imps, subst, decl =
+    new_instance_common ~program_mode ~generalize env instid ctx cl in
+  id, do_instance_interactive env env' sigma ?hook ~tac ~global ~poly ~program_mode
+    cty k u ctx ctx' pri decl imps subst id
+
+let new_instance_program ?(global=false) ~program_mode
```

Comment:
The way program_mode is passed here and in sibling functions seems strange.

Reply:
I agree, I need to do "constant propagation", this is the cleanup I was talking about...

Reply:
done